packages/vscode-extension/src/project/applicationContext.ts

```diff
@@ -0,0 +1,60 @@
+import { Disposable, window } from "vscode";
+import { BuildCache, migrateOldBuildCachesToNewStorage } from "../builders/BuildCache";
+import { DependencyManager } from "../dependency/DependencyManager";
+import { Logger } from "../Logger";
+import { LaunchConfigController } from "../panels/LaunchConfigController";
+import { findAppRootFolder } from "../utilities/extensionContext";
+import { Platform } from "../utilities/platform";
+import { setupPathEnv } from "../utilities/subprocess";
+
+export class ApplicationContext implements Disposable {
+  public appRootFolder: string;
+  public dependencyManager: DependencyManager;
+  public launchConfig: LaunchConfigController;
+  public buildCache: BuildCache;
+  private disposables: Disposable[] = [];
+
+  constructor() {
+    const newAppRoot = findAppRootFolder();
+    if (!newAppRoot) {
+      window.showErrorMessage(
+        "Failed to determine any application root candidates, you can set it up manually in launch configuration",
+        "Dismiss"
+      );
+      Logger.error("[Project] The application root could not be found.");
+      throw Error(
+        "Couldn't find app root folder. The extension should not be activated without reachable app root."
+      );
+    }
```

Comment:
NIT: I think I'd prefer if the `appRootFolder` was passed as an argument to the constructor to remove as much "logic" from it, but that's just my opinion.